packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -755,6 +776,14 @@ export const Table = <
             ...style,
           },
         })}
+        onKeyDown={(e) => {
+          if (e.shiftKey) {
```

Comment:
This will only fire when the table is focused, so it might be better to add it to the document.

Also we only want to disable text selection when the table has multi-selection enabled so it will need an extra condition.

Reply:
I added `keyup` and `keydown` event listeners to the ownerDocument (408-441 of below link). I'm not sure if this is what you meant of if there is a cleaner way to do it. So would be great if you can check how I implemented it [(commits diff)](https://github.com/iTwin/iTwinUI-react/pull/774/files/60657df4a537c23b43f5ff59d4231bcf3b74eda1..82731b8d342cb72acf887a84f7d54a23dfe4ff73).

Additionally, I also added a check to only run the shift select selection logic and to only run the disable selection while Shift is held logic when `selectionMode === 'multi'`. This addition is also in the above link commits diff link (612 of above link).